crates/bevy_pbr/src/light.rs

```diff
@@ -294,9 +294,26 @@ impl ClusterConfig {
             } => {
                 let aspect_ratio = screen_size.x as f32 / screen_size.y as f32;
                 let per_layer = *total as f32 / *z_slices as f32;
+                assert!(
+                    per_layer >= 1.0,
+                    "ClusterConfig has more z-slices than total clusters!"
+                );
+
                 let y = f32::sqrt(per_layer / aspect_ratio);
-                let x = (y * aspect_ratio).floor() as u32;
-                let y = y.floor() as u32;
+
+                let mut x = (y * aspect_ratio).floor() as u32;
+                let mut y = y.floor() as u32;
```

Comment:
Now I think about it, the `.floor()` should be unnecessary as casting to `u32` to should truncate.
```suggestion
                let mut x = (y * aspect_ratio) as u32;
                let mut y = y as u32;
```